src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -77,28 +81,44 @@ public interface Expression<T> extends SyntaxElement, Debuggable {
 	default Optional<T> getOptionalSingle(Event e) {
 		return Optional.ofNullable(getSingle(e));
 	}
-	
+
 	/**
 	 * Get all the values of this expression. The returned array is empty if this expression doesn't have any values for the given event.
 	 * <p>
 	 * The returned array must not contain any null values.
 	 * <p>
 	 * Do not use this in conditions, use {@link #check(Event, Checker, boolean)} instead.
 	 * 
-	 * @param e The event
+	 * @param event The event the expression is being used in.
 	 * @return An array of values of this expression which must neither be null nor contain nulls, and which must not be an internal array.
 	 */
-	public T[] getArray(final Event e);
-	
+	public T[] getArray(Event event);
+
+	/**
+	 * Get all the values of this expression and then converts them using the provided converter.
+	 * <p>
+	 * This method will use {@link Expression#getArray(Event)} to use non null values in the Converter.
+	 * 
+	 * @param <G> The returning type of the converter.
+	 * @param event The event the expression is being used in.
+	 * @param converter The converter to do the convertering to the type.
+	 * @return The values completed by the converter.
+	 */
+	@SuppressWarnings("serial")
+	default <G> G[] getArray(Event event, Converter<T, G> converter) {
+		assert converter != null;
+		return Converters.convertUnsafe(getArray(event), new TypeToken<G>(converter.getClass()){}.getRawType(), converter);
```

Comment:
I've attached an example of how this could be done without TypeToken by using ConverterInfo to get the result Class of the converter.

```suggestion
	@SuppressWarnings("serial")
	default <G> G[] getArray(Event event, ConverterInfo<T, G> converterInfo) {
		return Converters.convertUnsafe(getArray(event), converterInfo.to, converterInfo.converter);
	}
```

Lambda usage is still possible, with a little additional (albeit necessary) boilerplate:
```java
Location[] locations = getArray(event, new ConverterInfo(Block.class, Location.class, Block::getLocation, NO_LEFT_CHAINING));
//                                                                                    ^ look mom a method reference!
```

Reply:
This is certainly an alternative :) (although I don't think it's necessary to specify any flags)

I do wonder if this is any better than just having another parameter for Class<G>

Reply:
> This is certainly an alternative :) (although I don't think it's necessary to specify any flags)
> 
> I do wonder if this is any better than just having another parameter for Class

It is actually necessary to specify at least `0` for the flags, as all ConverterInfo constructors require it...
![Project Structure in Intellij IDEA, showing constructors of ConverterInfo, each requiring `int` options parameter](https://github.com/SkriptLang/Skript/assets/52505120/9f861589-b3fd-4cb2-9f81-69c340837f75)

Definitely a <kbd>good first issue</kbd> for someone :)

Having a `Class<G>` parameter actually seems like a better appraoch  — it's something I hadn't considered.

1. No need for the `0` in the ConverterInfo constructor
2. Neater lambda syntax:

    ```diff
    - Location[] locations = getArray(event, new ConverterInfo(Block.class, Location.class, Block::getLocation, 0));
   + Location[] locations = getArray(event, Location.class, Block::getLocation);
    ```
3. Mirrors the signature of `Converters.convert`, essentially a short-hand

We could even get rid of the `unsafe` and just use `Converters.convert` (though I think we can also do that with ConverterInfo):

```suggestion
	@SuppressWarnings("serial")
	default <G> G[] getArray(Event event, Class<G> to, Converter<T, G> converter) {
		assert converter != null;
		return Converters.convert(getArray(event), to, converter);
```

Reply:
As stated again, no one has Skript implemented ConverterInfo as a public common access for the API, they're going to have to collect that in the first place, and as for adding a class parameter, there is no need to add an un-needed parameter when TypeToken accomplishes exactly what's needed without the need of both formentioned.

Reply:
It's not beta anymore as expressed in a comment from TP. That's old Guava versions. Spigot or Mojang probably finally updated it to latest versions when they forced 1.17 onto people. TypeToken is literally the central heart of GSON, Google uses it frequently.